### PR TITLE
Add some defines and structs required by cmake

### PR DESCRIPTION
**Pull request by Dennisbonke**
This PR adds 4 network related constants, 2 ioctl constants and 2 structs, all shamelessly borrowed from my host.

### CODE REVIEW

File: abis/mlibc/in.h [hunk 1]
```diff
@@ -53,6 +53,19 @@ struct ip_mreq {
 	struct in_addr imr_interface;
 };
 
+// These structs are copied from linux and modified to work, they may be severely wrong
```
Comment (Geertiebear) [inline abis/mlibc/in.h:56]: Why did they need to be "modified to work"?

Comment (Geertiebear) [reply abis/mlibc/in.h]: Do these structs match linux now/are they not modified to work anymore?

**Comment by Geertiebear** [comment]
Adding the struct to the linux ABI in a seperate commit leaves a broken commit behind, if you could squash it into the previous commit.

File: abis/mlibc/in.h (hunk 1 above)
Comment (Dennisbonke) [reply abis/mlibc/in.h]: yes, these do match linux now

**Review by Geertiebear — APPROVED** [review]
LGTM now

**Pull request merged**